lib/configuration.sh

```diff
@@ -548,29 +487,19 @@ fi
 
 # Build final package list after possible override
 PACKAGE_LIST="$PACKAGE_LIST $PACKAGE_LIST_RELEASE $PACKAGE_LIST_ADDITIONAL"
-
-# Clean up the list
-# Myy : Replace any single or multiple instance of 'space' characters
-# (spaces, tabs, newlines) by a single space character
-# DO NOT PUT quotes after 'echo', else the trick won't work.
-PACKAGE_LIST="${PACKAGE_LIST#"${PACKAGE_LIST%%[![:space:]]*}"}"
-PACKAGE_LIST="${PACKAGE_LIST%"${PACKAGE_LIST##*[![:space:]]}"}"
-PACKAGE_LIST="$(echo ${PACKAGE_LIST})"
-
-PACKAGE_MAIN_LIST="${PACKAGE_LIST}"
-
-# FIXME Myy: Factorize this...
-PACKAGE_LIST_DESKTOP="${PACKAGE_LIST_DESKTOP#"${PACKAGE_LIST_DESKTOP%%[![:space:]]*}"}"
-PACKAGE_LIST_DESKTOP="${PACKAGE_LIST_DESKTOP%"${PACKAGE_LIST_DESKTOP##*[![:space:]]}"}"
-PACKAGE_LIST_DESKTOP="$(echo ${PACKAGE_LIST_DESKTOP})"
+PACKAGE_MAIN_LIST="$(cleanup_list PACKAGE_LIST)"
 
 [[ $BUILD_DESKTOP == yes ]] && PACKAGE_LIST="$PACKAGE_LIST $PACKAGE_LIST_DESKTOP"
+PACKAGE_LIST="$(cleanup_list PACKAGE_LIST)"
 
 # remove any packages defined in PACKAGE_LIST_RM in lib.config
+# FIXME : This actually nuke any content set to PACKAGE_LIST_RM
```

Comment:
You mean it doesn't work if set by user otherwise it works?

Reply:
Well, that means that if you define any PACKAGE_LIST_RM variable, it will be overwritten by this command.

The right way, IMHO, is to do this `aggregated_content="${PACKAGE_LIST_RM} "`. That way, this the variable should be preserved.
I'll try this fix tonight.